src/lib.rs

```diff
@@ -60,8 +60,17 @@ impl AbsPathBuf {
     /// Convert from a known absolute path.
     ///
     /// Passing a relative path is a programming bug (unchecked).
-    pub fn new_unchecked(path: PathBuf) -> Self {
-        AbsPathBuf(path)
+    pub fn new_unchecked(path: &PathBuf) -> Self {
+        Self::new_unchecked_normalized(path)
+    }
+
+    fn new_unchecked_normalized(path: &Path) -> Self {
+        let mut normalized = PathBuf::new();
+        // I didn’t find a better way to normalize a path (remove double `/` and `/./` and the like)
```

Comment:
The only thought I have is `PathBuf.canonicalize` - IIRC when I tried using that it threw errors if the path wasn't real though.